hphp/hack/src/typing/typing.ml

```diff
@@ -247,10 +247,15 @@ let expr_any env p e =
   (env, with_type ty Tast.dummy_saved_env e, ty)
 
 let unbound_name env (pos, name) e =
+  let class_exists =
+    match Env.get_class env name with
```

Comment:
This API fetches any 'classish', so `name` might be an interface or trait. We shouldn't offer `new` for things that aren't classes.